ecommerce/views.py

```diff
@@ -152,3 +164,54 @@ def post(self, request, *args, **kwargs):
                     )
         # The response does not matter to CyberSource
         return Response()
+
+
+class CouponsView(ListModelMixin, GenericViewSet):
+    """
+    View for coupons API. This is a read-only API showing the user
+    - what coupons they have available if those coupons would be automatically applied on checkout
+    - what coupons they have for a given coupon code, even if those coupons wouldn't be automatically applied
+    """
+    authentication_classes = (
+        SessionAuthentication,
+        TokenAuthentication,
+    )
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        """List coupons which a user is allowed to see"""
+        return [
+            coupon for coupon in Coupon.objects.all()
+            if is_coupon_redeemable(coupon, self.request.user) and (
+                coupon.is_automatic or UserCoupon.objects.filter(user=self.request.user, coupon=coupon).exists()
+            )
+        ]
```

Comment:
On a related note, this API will only return zero or one coupons. I talked to @pdpinch and we need to make a decision somewhere which coupon to display. I'll add a function in `api.py` to calculate that, and try to do filtering to avoid n+1 queries